server/src/main/java/org/apache/iotdb/db/query/dataset/RawQueryDataSetWithValueFilter.java

```diff
@@ -118,7 +118,7 @@ private boolean cacheRowRecords() throws IOException {
 
       // 3. use values in results to fill row record
       for (int j = 0; j < cachedTimeCnt; j++) {
-        if (results[j] == null) {
+        if (results == null || results[j] == null) {
```

Comment:
It seems that only when `cachedTimeCnt` equals to zero, `getValuesInTimestamps` will return null. However, if `cachedTimeCnt` equals to zero, we will `return false` before here.